tools/environment/export.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"golang.org/x/exp/maps"
+)
+
+type exporter struct {
+	root string
+}
+
+func (ex *exporter) Export() error {
+	fmt.Println("* Exporting environment...")
+	files, err := descopeClient.Management.Environment().ExportRaw()
+	if err != nil {
+		return err
+	}
+
+	if Flags.Debug {
+		WriteDebugFile(ex.root, "debug/export.log", files)
+	}
+
+	fmt.Println("* Writing files...")
+	paths := maps.Keys(files)
+	sort.Strings(paths)
+	for _, path := range paths {
+		fmt.Printf("  - %s\n", path)
+		data := files[path]
+		if object, ok := data.(map[string]any); ok {
+			if err := ex.writeObject(path, object); err != nil {
+				return fmt.Errorf("error writing json file %s: %w", path, err)
+			}
+		} else if asset, ok := data.(string); ok {
+			if err := ex.writeAsset(path, asset); err != nil {
+				return fmt.Errorf("error writing asset file %s: %w", path, err)
+			}
+		} else {
+			return fmt.Errorf("unexpected file data in exported project: %s", path)
+		}
+	}
+
+	fmt.Println("* Done")
+
+	return nil
+}
+
+func (ex *exporter) writeObject(path string, object map[string]any) error {
+	bytes, err := json.MarshalIndent(object, "", "  ")
+	if err != nil {
+		return err
+	}
+	return ex.writeBytes(path, bytes)
+}
+
+func (ex *exporter) writeAsset(path string, asset string) error {
+	if filepath.Ext(path) == ".txt" || filepath.Ext(path) == ".html" {
```

Comment:
I wonder if sematically we should do the oposite here - e.g. decode only if `filepath.Ext(path)` is jpg/png/...


this will also enable us to maintain this list of extension once

![image](https://github.com/descope/go-sdk/assets/10514677/fc479e06-1f04-4380-b79d-61fed5f992ff)

wdyt?

Reply:
I can refactor it somehow, no problem, but in general it's much more likely that we'll add more binary types than anything something other than `.txt`/`.html` as plain text, so I preferred having that as the special case.